File: APIs/ChatService/ChatService/DTOs/AddUserToChatGroupModel.cs
using Newtonsoft.Json;

namespace ChatService.DTOs
{
    public class AddUserToChatGroupModel
    {
        [JsonProperty("userWhoAdded")]
        public int UserWhoAdded { get; set; }
        [JsonProperty("newGroupPartnerList")]
        public List<UserObject> NewGroupPartnerList { get; set; } = null!;
        [JsonProperty("chatListItem")]
        public LastMessage ChatListItem { get; set; } = null!;
        [JsonProperty("addedUsers")]
        public List<UserObject> AddedUsers { get; set; } = null!;
    }
}

File: SocialAppFlutter/social_app/lib/Models/add_user_to_chat_group_model.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:social_app/Models/last_message.dart';
import 'package:social_app/Models/user.dart';

part 'json_generated_code/add_user_to_chat_group_model.g.dart';

@JsonSerializable(explicitToJson: true)
class AddUserToChatGroupModel {
  final int userWhoAdded;
  final List<User> newGroupPartnerList;
  final LastMessage chatListItem;
  final List<User> addedUsers;

  AddUserToChatGroupModel(
      this.userWhoAdded, this.newGroupPartnerList, this.chatListItem, this.addedUsers);

  factory AddUserToChatGroupModel.fromJson(Map<String, dynamic> json) =>
      _$AddUserToChatGroupModelFromJson(json);

  Map<String, dynamic> toJson() => _$AddUserToChatGroupModelToJson(this);
}
